src/core/a-entity.js

```diff
@@ -23,155 +22,161 @@ var ONCE = {once: true};
-var proto = Object.create(ANode.prototype, {
-  createdCallback: {
-    value: function () {
-      this.components = {};
-      // To avoid double initializations and infinite loops.
-      this.initializingComponents = {};
-      this.componentsToUpdate = {};
-      this.isEntity = true;
-      this.isPlaying = false;
-      this.object3D = new THREE.Group();
-      this.object3D.el = this;
-      this.object3DMap = {};
-      this.parentEl = null;
-      this.rotationObj = {};
-      this.states = [];
-    }
-  },
+class AEntity extends ANode {
+  constructor () {
+    super();
+    this.components = {};
+    // To avoid double initializations and infinite loops.
+    this.initializingComponents = {};
+    this.componentsToUpdate = {};
+    this.isEntity = true;
+    this.isPlaying = false;
+    this.object3D = new THREE.Group();
+    this.object3D.el = this;
+    this.object3DMap = {};
+    this.parentEl = null;
+    this.rotationObj = {};
+    this.states = [];
+  }
 
   /**
    * Handle changes coming from the browser DOM inspector.
    */
-  attributeChangedCallback: {
-    value: function (attr, oldVal, newVal) {
-      var component = this.components[attr];
-      // If the empty string is passed by the component initialization
-      // logic we ignore the component update.
-      if (component && component.justInitialized && newVal === '') {
-        delete component.justInitialized;
-        return;
-      }
-      // When a component is removed after calling el.removeAttribute('material')
-      if (!component && newVal === null) { return; }
-      this.setEntityAttribute(attr, oldVal, newVal);
+  attributeChangedCallback (attr, oldVal, newVal) {
+    var component = this.components[attr];
+
+    super.attributeChangedCallback();
+    // If the empty string is passed by the component initialization
+    // logic we ignore the component update.
+    if (component && component.justInitialized && newVal === '') {
+      delete component.justInitialized;
+      return;
     }
-  },
+    // When a component is removed after calling el.removeAttribute('material')
+    if (!component && newVal === null) { return; }
+    this.setEntityAttribute(attr, oldVal, newVal);
+  }
 
   /**
-   * Add to parent, load, play.
-   */
-  attachedCallback: {
-    value: function () {
-      var assetsEl;  // Asset management system element.
-      var sceneEl = this.sceneEl;
-      var self = this;  // Component.
+  * Add to parent, load, play.
+  */
+  connectedCallback () {
+    // Defer if DOM is not ready.
+    if (document.readyState === 'loading') {
+      document.addEventListener('DOMContentLoaded', this.connectedCallback.bind(this));
+      return;
+    }
 
-      this.addToParent();
+    AEntity.prototype.doConnectedCallback.call(this);
+  }
 
-      // Don't .load() scene on attachedCallback.
-      if (this.isScene) { return; }
+  doConnectedCallback () {
+    var assetsEl;  // Asset management system element.
+    var sceneEl = this.sceneEl;
```

Comment:
I confirm, I just added a console.log, `this.sceneEl` is always undefined here. I you put `super.connectedCallback();` above, it will contain the scene element.